CMakeLists.txt

```diff
@@ -16,6 +16,7 @@ SET(PROJECT_DESCRIPTION "Rigid multi body dynamics algorithms")
 SET(PROJECT_URL "http://github.com/stack-of-tasks/pinocchio")
 SET(PROJECT_CUSTOM_HEADER_EXTENSION "hpp")
 OPTION(INSTALL_DOCUMENTATION "Generate and install the documentation" ON)
+OPTION(BUILD_EXAMPLES "Build and test the examples" ON)
```

Comment:
I would set it OFF by default

Reply:
Why ?

Reply:
These are examples. They do not add any functionality, so I see no point in building them by default. I think the best approach is that whoever wants them should explicitly ask for them, otherwise nothing. Pinocchio already takes a lot of time to compile, let's not add by default an unnecessary other burden that needs to be switched off if you just want Pinocchio to be working and you want to keep compilation times below the span of a human life

Reply:
> I see no point in building them by default.
These are tests. The point in building them is to check that the examples are still up to date.

>  think the best approach is that whoever wants them should explicitly ask for them, otherwise nothing
Then are you also asking to turn `BUILD_UNIT_TESTS` to `OFF` by default ?

> Pinocchio already takes a lot of time to compile
```
$ cmake .. ; time make -j8 ; time make test
0:32.21elapsed 621%CPU
0:05.32elapsed 319%CPU

$ cmake -DBUILD_EXAMPLES=OFF .. ; time make -j8 ; time make test
0:30.50elapsed 619%CPU
0:05.48elapsed 299%CPU

$ cmake -DBUILD_UNIT_TESTS=OFF ..; time make -j8
0:16.67elapsed 107%CPU
```

Once again, are you asking to turn `BUILD_UNIT_TESTS` to `OFF` by default ?.

If you let `BUILD_EXAMPLES` to `ON` but put `BUILD_UNIT_TESTS` to `OFF`, they will not be compiled. 
Therefore, I think we should remove this `BUILD_EXAMPLES` option, it really adds nothing, and if someone really want to turn this compilation off, they might as well just disable all tests.

Reply:
I do agree.

Reply:
if you want me to remove ccache, I can run those benchmarks again, but my point is that no, for now, we have like ~50 loc for 2 .cpp examples with only one main function. The compilation time of that is neglegible.

Reply:
@jcarpent thanks, fixed in 7b69be675d8cf42b1e1d1f9f456b959607ff620e

Reply:
Sounds good to me.